A3A/addons/core/functions/Missions/fn_LOG_Crashsite.sqf

```diff
@@ -0,0 +1,861 @@
+        ||
+	    _box distance _deliverySite < 50 
+        ||
+	    _box distance (getMarkerPos respawnTeamPlayer) < 50
+        ||
+        _box distance (getMarkerPos traderMarker) < 50
+        ||
+	    dateToNumber date > _dateLimitNum
+    };
+} else {
+    waitUntil {
+	    sleep 1;
+	    !alive _box 
+        ||
+	    _box distance _deliverySite < 50 
+        ||
+	    _box distance (getMarkerPos respawnTeamPlayer) < 50
+        ||
+	    dateToNumber date > _dateLimitNum
+    };
+};
+switch(true) do {
+    case(_box distance _deliverySite < 50 || {dateToNumber date > _dateLimitNum}): {
+        Info("Box has been recovered by enemy, mission falied.");
+
+        [_taskId, "LOG", "FAILED"] call A3A_fnc_taskSetState;
+        [_taskId2, "LOG", "FAILED"] call A3A_fnc_taskSetState;
+
+        [-900, _sideX] remoteExec ["A3A_fnc_timingCA",2];
+        [-10,theBoss] call A3A_fnc_addScorePlayer;
+        if (dateToNumber date > _dateLimitNum) then {
+		_hrT = server getVariable "hr";
+		_resourcesFIAT = server getVariable "resourcesFIA";
+		[-1*(round(_hrT/3)),-1*(round(_resourcesFIAT/3))] remoteExec ["A3A_fnc_resourcesFIA",2];
+		[-10*_factor, 90] remoteExec ["SCRT_fnc_rivals_reduceActivity",2];
+		{ A3A_curHQInfoInv = A3A_curHQInfoInv + 0.25 + random 0.5 } remoteExecCall ["call", 2];
+	    }; ///If players fail, enemy will get location of the HQ
+    };
+    case(!alive _box): {
+        Info("Box has been destroyed, mission canceled.");
+        [_taskId, "LOG", "CANCELED"] call A3A_fnc_taskSetState;
+        [_taskId2, "LOG", "CANCELED"] call A3A_fnc_taskSetState;
+        [-300, _sideX] remoteExec ["A3A_fnc_timingCA",2];
+    };
+    case(_box distance (getMarkerPos respawnTeamPlayer) < 50): {
+        Info("Box has been delivered to HQ, mission completed.");
+        [_taskId, "LOG", "SUCCEEDED"] call A3A_fnc_taskSetState;
+        [_taskId2, "LOG", "CANCELED"] call A3A_fnc_taskSetState;
+
+        [0, 600] remoteExec ["A3A_fnc_resourcesFIA",2];
+        [1800, _sideX] remoteExec ["A3A_fnc_timingCA",2];
+        { 
+            [20*_bonus,_x] call A3A_fnc_addScorePlayer;
+            [350*_bonus,_x] call A3A_fnc_addMoneyPlayer;
+        } forEach (call SCRT_fnc_misc_getRebelPlayers);
+        [10*_bonus,theBoss] call A3A_fnc_addScorePlayer;
+        [250*_bonus,theBoss, true] call A3A_fnc_addMoneyPlayer;
+        ["Large", _sideX] remoteExec ["A3A_fnc_selectIntel", 2];
+        [(position _box), 4000, 1200, false] spawn SCRT_fnc_common_recon; ///params ["_position", "_radius", "_revealTime", ["_isInterrogation", false]]; ///revels enemy location only once, which is not good?, dunno maybe it should do it every 2-5 minutes
+        sleep 60;
+        [(position _box), 4000, 1200, false] spawn SCRT_fnc_common_recon;
+        sleep 60;
+        [(position _box), 4000, 1200, false] spawn SCRT_fnc_common_recon;
+        sleep 60;
+        [(position _box), 4000, 1200, false] spawn SCRT_fnc_common_recon;
+        deleteVehicle _box;

```

Comment:
"What the absolute **** is this?!!?"
Sincerely,
The scheduler

This will need to be changed in some way, shape or form. It's just awful.
Perhaps use a for loop?
```suggestion
        for "_i" from 0 to 3 do {
          [(position _box), 4000, 1200, false] spawn SCRT_fnc_common_recon;
          uiSleep 60;
        };        
        deleteVehicle _box;
```
Make sure to test it before merging change

Reply:
yep, that works